ste/uploader-blobFS.go

```diff
@@ -166,14 +218,17 @@ func (u *blobFSUploader) GenerateUploadFunc(id common.ChunkID, blockIndex int32,
 			jptm.FailActiveUpload("Uploading range", err)
 			return
 		}
+
+		u.tp[blockIndex].Completed = true
```

Comment:
I'm not particularly comfortable with one GR writing Completed and another reading it.   It's really straying outside of what's consider safe multi-threading (multi GR) practice. In this case, I think we'd probably get away with it, because the reader would eventually read the right thing, even if it does't read it quite as soon as we expect.  But... I'm be a lot more comfortable if we had 1 GR per file; and what we pushed to the channel was the  id.OffsetInFile, and then the GR that reads the channnel would maintain sate however it whished (e.g. probabl3y with an array that it has as a local variable).

Reply:
Correction, I think we would be OK, but only because we'd be relying on a side effect of channel operations that they would, in effect, guarantee the thread safety of .Completed. But, the Go documentation on that page suggests that, if you have to rely on that, you probably shouldn't.